README.md

```diff
@@ -1,3 +1,128 @@
 # amqp
 
-Simplified publish/subscribe and connection retry handling for amqp (rabbitmq)
+Simplified publish/subscribe and connection retry handling for amqp (rabbitmq).
+
+This package integrates with [amqplib](http://www.squaremobius.net/amqp.node/channel_api.html) and [amqp-connection-manager](https://github.com/benbria/node-amqp-connection-manager) (which is also built on amqplib).  Use one of those packages to create connections that can be used with this package.
+
+## Install
+
+```
+npm install @nowait/amqp --save
+```
+
+## Examples
+
+### Create a Channel
+
+First, create an AMQP connection, using either amqplib or amqp-connection-manager.  Then, create a channel.
+
+#### Create duplex channel
+
+Duplex channels are provided by amqplib, and can be used for either publishing or consuming messages.  They don't provide any fault tolerance for publishers: when connections are dropped, publishers will fail.
+
+```js
+import amqp from 'amqplib'
+import { createChannel } from '@nowait/amqp'
+
+// See amqplib documentation
+// http://www.squaremobius.net/amqp.node/channel_api.html#connect
+const connection = amqp.connect(uri, options)
+
+const channel = createChannel(connection)
+```
+
+#### Create managed channel
+
+Managed channels can only be used for event *publishers*.  They provide fault-tolerance by buffering events when the connection is dropped.  It will attempt to reconnect, and then deliver the buffered events.
+
+```js
+import amqpcm from 'amqp-connection-manager'
+import { createManagedChannel } from '@nowait/amqp'
+
+// See amqp-connection-mananger documentation
+// https://github.com/benbria/node-amqp-connection-manager#basics
+const connection = amqpcm.connect([...uri], options)
+
+const channel = createManagedChannel(connection)
+```
+
+### Publish messages to a queue
+
+After creating a Channel, use `publishTo` to publish messages to a queue.
+
+```js
+// Assuming channel was created using one of the approaches above
+
+const publishChannel = publishTo(channel)
+
+const queueConfig = {
+  exchangeName: 'some-exchange',
+  queueName: 'some-queue',
```

Comment:
@briancavalier We talked about this yesterday, but I'm not seeming to follow this code. This option exists in publishTo and assertChannel also asserts the queueName. But is it optional? For a publish/subscribe pattern (or at least the ones I've seen) each subscriber consumes from its own queue. You don't really want to specify a queueName upon publishing otherwise you'd have to specify for each consumer. The publishers are better off being agnostic to the subscribers and queueNames.

If it is optional, where in the code can I tell that?